seed/redis/config.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 HUMAN Security.
+// Use of this source code is governed by a MIT style
+// license that can be found in the LICENSE file.
+
+package redis
+
+import "github.com/go-redis/redis/v8"
+
+// SeedConfig represents the configuration for the redis seed component.
+type SeedConfig struct {
+	// Address - a valid redis server address to connect to
+	Address string `json:"address,omitempty"`
+
+	// ClientProvider - can be used as an alternative to Address, provides a redis client to use.
+	// available only via code, not available in config files.
+	// if both ClientProvider and Address are provided, ClientProvider is used.
+	ClientProvider func() (*redis.Client, error) `json:"-"`
+
+	// Data - a list of objects, each represents a redis key and its data
+	Data []*SeedData `json:"data,omitempty"`
+}
+
+// SeedData represents data for a redis hash.
+type SeedData struct {
+	// Key - the name of the redis key
+	Key string `json:"key,omitempty"`
```

Comment:
my only comment on the config is that this structure will only let us support HSet, or at least block the name `Key` for this specific use.
What do you think about something like this:
```go
type SeedData struct {
	SetKeys  []*Set  `json:"set_keys"`
	HSetKeys []*HSet `json:"hset_keys"`
}

type Set struct {
	Key   string        `json:"key"`
	Value string        `json:"value"`
	TTL   time.Duration `json:"ttl"`
}

type HSet struct {
	Key    string            `json:"key"`
	Values map[string]string `json:"values"`
	TTL    time.Duration     `json:"ttl"`
}
```

this will allow us to add future support for other data types as well without breaking anything
WDYT?

Reply:
Looks good